pyproject.toml

```diff
@@ -41,7 +41,7 @@ python = "^3.9,<3.11"
 
 h5py = "^3.2.1"
 numpy = "!=1.19.4, ~1.23.5"
-Pillow = "~10.1.0"
+Pillow = "^10.3.0"
```

Comment:
**thought:** specifying `^10.3.0` allows any `10.x` version, while `~10.3.0` is any `10.3.x`

In the past, updating to a minor version broke the behavior at runtime. I would tend to limit to `~10.3.0`
Alos relocking  a repo with `^10.3.0` spec might lock on `10.5`, while a repo which has not been relocked yet will still use `10.3.0`. that create more discrepancy between repos.

it is quite depends on packages though. Some have more conservative policy on changing minor version.